### DISCUSSION TITLE
auto-increment unique fields

### DISCUSSION THREAD

**Discussion by DanRibbens** [payloadcms/payload#1026] [Roadmap]
When duplicating or creating new documents with similar unique fields, it can be difficult for the editor. An enhancement to solve this problem is to add a configuration option on fields that are unique to increment them automatically on the backend, before validation. This would allow a duplicated collection to have a text field for example being saved as `value` to be automatically changed to `value-1` for example, instead of failing with a validation error when `value` is already present.

This could work the same with number field types as well.

The config option: `autoIncrement` needs to be available for field types `text` and `number`, only when `unique` is true.

**Comment by thgh** [comment]
[Here is some inspiration](https://github.com/payloadcms/payload/issues/1243#issuecomment-1308992274)

**Comment by Snailedlt** [comment]
Here's an alternative solution for duplicating documents: #6157

**Comment by DanRibbens** [comment]
This has been mostly handled in v3. I rewrote duplicate to be done on the server side and when I did so I added default beforeDuplicate field level hooks for unique fields.

This probably makes it not necessary to add a separate flag for the `autoIncrement` since text is handled for you in part and anything you need to customize, you can with your own beforeDuplicate hook function.

**Reply by Snailedlt** [reply]
I still think it would be much better to not save the duplicated document when the user clicks on duplicate.

It's very rare that you want to duplicate without changing any fields. If it is saved automatically (like it is now), the duplicated document will be shown on the production site before it is finished, sort of in a limbo state between duplicating, and editing necessary fields.

It would be much better if the document data was copied over without saving, and the user was left with the decision on when to save it themselves.

**Reply by thgh** [reply]
Having a route `/admin/collections/widgets/create?duplicate=idabc123` would allow errors to be handled client-side. It's hard for developers to foresee every possible error.

What is the reasoning for persisting duplicated documents? Saving a click in case the user actually wants a pure copy?

**Reply by DanRibbens** [reply]
I agree that the UX and even the DX would be improved to do it this way. The approach is complicated by having to support multiple locales.

**Comment by JarrodMFlesch** [comment]
The duplicate as draft was implemented in [this PR](https://github.com/payloadcms/payload/pull/14619).
